src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1229,23 +1233,144 @@ private int SkipBlockComment(int i)
             return i;
         }
 
-        private static char Backtick(char c)
+        private string Backtick(char c)
         {
             switch (c)
             {
-                case '0': return '\0';
-                case 'a': return '\a';
-                case 'b': return '\b';
-                case 'e': return '\u001b';
-                case 'f': return '\f';
-                case 'n': return '\n';
-                case 'r': return '\r';
-                case 't': return '\t';
-                case 'v': return '\v';
-                default: return c;
+                case '0': return "\0";
+                case 'a': return "\a";
+                case 'b': return "\b";
+                case 'e': return "\u001b";
+                case 'f': return "\f";
+                case 'n': return "\n";
+                case 'r': return "\r";
+                case 't': return "\t";
+                case 'u': return ScanUnicodeEscapeSequence();
+                case 'v': return "\v";
+                default: return c.ToString();
             }
         }
 
+        private string ScanUnicodeEscapeSequence()
+        {
+            int escSeqStartIndex = _currentIndex - 2;
+            bool isBracketedSequence = false;
+            bool isTerminated = false;
+
+            char c = PeekChar();
+            if (c == '{')
+            {
+                SkipChar();
+                isBracketedSequence = true;
+            }
+
+            int maxNumberOfHexDigits = (isBracketedSequence ? 6 : 4);
+
+            // Scan hex chars after the Unicode escape sequence start.
+            var sb = GetStringBuilder();
+            int i;
+            for (i = 0; i < maxNumberOfHexDigits; i++)
```

Comment:
That's not the design.  The design is that there are two forms of Unicode escape sequence.  The first is a nod to C# and Bash (folks can take strings from either and replace `\` with `` ` `` and it just works e.g. `"\u00A9 My Company Name"` to `` "`u00A9 My Company Name"``.  This fixed length is why we stop the loop at four for this form because `` "`u0041bcde"`` outputs `Abcde` so the `bcde` part, while valid hex digits are **not** part of the Unicode escape sequence.

The second form is a variable length form that takes 1 to 6 hex digits in braces (inspired by JavaScript) allows for Unicode characters above the bilingual multiplane (BMP).  IOW folks can insert emoji characters with a sequence like `` "`u{1F44D}"``.

This discussion on this feature is here: https://github.com/PowerShell/PowerShell/issues/2751